### ISSUE TITLE
BLangNodeTransformer throws CCE for incomplete sources.

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#27666]
**Description:**
Consider the following source snippet,

```
type TestType client object {
    # Test Description1 for test function
    #
    # + a - a Parameter Description1 
    # + return - Return Value Description1
    function testFunction(int a) returns int; // 1

    # Test Description for remote function1
    #
    # + a - a Parameter Description1 
    # + return - Return Value Description1
    remote function testRemoteFunction1(int a) returns int;
};

client class TestClass {
    *TestType;
    # Test Description2 for test function
    #
    # + a - a Parameter Description2 
    # + return - Return Value Description2
    function testFunction(int a) returns int { // 2
        return 123;
    }

    # Test Description2 for remote function1
    #
    # + a - a Parameter Description2 
    # + return - Return Value Description2
    remote function testRemoteFunction1(int a) returns int { // 2
        return 123;
    }

    # Test Description for remote function2
    #
    # + b - b Parameter Description1 
    # + return - Return Value Description1
    remote function testRemoteFunction2(int b) returns int { // 2
        return 123;
    }
}

public function main() {
    TestClass c = new();
    int var1 = c.testFunction(1); // 3
    int var2 = c->
}
```
Above source throws the following CCE

```
at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createExpression(BLangNodeTransformer.java:4399)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createBLangVarDef(BLangNodeTransformer.java:2702)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2669)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:442)
	at io.ballerina.compiler.syntax.tree.VariableDeclarationNode.apply(VariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.generateAndAddBLangStatements(BLangNodeTransformer.java:4308)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1473)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:442)
	at io.ballerina.compiler.syntax.tree.FunctionBodyBlockNode.apply(FunctionBodyBlockNode.java:59)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.populateFunctionNode(BLangNodeTransformer.java:1381)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createFunctionNode(BLangNodeTransformer.java:1361)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1339)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:442)
	at io.ballerina.compiler.syntax.tree.FunctionDefinitionNode.apply(FunctionDefinitionNode.java:71)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:549)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:442)
	at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:472)
	at io.ballerina.projects.DocumentContext.compilationUnit(DocumentContext.java:111)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:330)
	at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:297)
	at io.ballerina.projects.ModuleCompilation.compile(ModuleCompilation.java:100)
	at io.ballerina.projects.ModuleCompilation.<init>(ModuleCompilation.java:62)
	at io.ballerina.projects.PackageContext.lambda$getModuleCompilation$0(PackageContext.java:162)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
	at io.ballerina.projects.PackageContext.getModuleCompilation(PackageContext.java:161)
	at io.ballerina.projects.Module.getCompilation(Module.java:97)
```

**Comment by lochana-chathura** [comment]
Recovers as,  `int var2 = c-> MISSING[default] MISSING[;]` (async-send-action)

Blocked by #24849

**Comment by lochana-chathura** [comment]
Duplicate of #24944

**Comment by rdhananjaya** [comment]
Closing as this is a duplicate of #24944
